internal/lsp/server.go

```diff
@@ -701,7 +701,7 @@ func (l *LanguageServer) handleTextDocumentCompletion(
 	}
 
 	return types.CompletionList{
-		IsIncomplete: false,
```

Comment:
This is a key part of the change here which has the client request completions on each char change. This allows the server to define the completions and filtering them.